### ISSUE TITLE
Splunk enterprise fact hangs partial install

### ISSUE THREAD

**Issue by rbclark** [voxpupuli/puppet-splunk#238]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 5.5.7
- Ruby: 2.4.4p296
- Splunk: 7.2.6
- Distribution: RedHat 7
- Module version: 3324eeb6626646523a29b6fb89af757c4cc22113

## How to reproduce (e.g Puppet code you use)

```
  class { '::splunk::enterprise':
    package_provider      => 'yum',
    manage_package_source => false,
  }
```

## What are you seeing

If the Splunk package is installed but the license is not accepted puppet hangs on every subsequent run on the command `Debug: Facter: executing command: /bin/sh -c /opt/splunk/bin/splunk --version`. This is due to that the version command is returning the Splunk license and hanging indefinitely.

## What behaviour did you expect instead

Puppet to still be able to run successfully even though Splunk is partially installed

## Output log

    Debug: Facter: executing command: /bin/sh -c /opt/splunk/bin/splunk --version

**Comment by posteingang** [comment]
I know the problem also. It happens if the license aggrement not finished.
I fixed this for me here: https://github.com/posteingang/puppet-splunk/blob/master/lib/facter/splunk_version.rb

**Comment by rbclark** [comment]
Yeah that is a lot more reasonable of a method, @posteingang have you put in a pull request to get this change merged in?

**Comment by posteingang** [comment]
Not yet... I am not finished with my tests.

**Comment by mwhitake** [comment]
Just ran into the same issue with facter hanging trying to determine the splunk version when the license has not been accepted.  This feels related to the fact that the license is not accepted when doing an upgrade rather than a new install.

**Comment by wcooley** [comment]
Another option is to add `--no-prompt` to the `bin/splunk --version` commands. At least, this works with Splunk Forwarder 7.3.3--I don't know how far back or after it works. (This causes the command to exit 2, so be prepared for that too.)

**Comment by root-expert** [comment]
This was fixed in 5778673eee9fa92453fda520cfafe23651abaac1
